pkg/api/v1alpha1/cluster.go

```diff
@@ -95,6 +95,13 @@ func WithClusterEndpoint() ClusterGenerateOpt {
 	}
 }
 
+// WithCPUpgradeRolloutStrategy allows add UpgradeRolloutStrategy option to cluster config under ControlPlaneConfiguration.
+func WithCPUpgradeRolloutStrategy(maxSurge int, maxUnavailable int) ClusterGenerateOpt {
+	return func(c *ClusterGenerate) {
+		c.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy = &ControlPlaneUpgradeRolloutStrategy{Type: "RollingUpdate", RollingUpdate: ControlPlaneRollingUpdateParams{MaxSurge: maxSurge}}
+	}
```

Comment:
nit: match the indentation in `WithWorkerMachineUpgradeRolloutStrategy()`
```suggestion
		c.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy = &ControlPlaneUpgradeRolloutStrategy{
		    Type: "RollingUpdate",
		    RollingUpdate: ControlPlaneRollingUpdateParams{MaxSurge: maxSurge},
		}
	}
```